typing/typecore.ml

```diff
@@ -2738,6 +2738,26 @@ let rec is_inferred sexp =
   | Pexp_ifthenelse (_, e1, Some e2) -> is_inferred e1 && is_inferred e2
   | _ -> false
 
+
+(* check if the type of %apply or %revapply matches the type expected by
+   the specialized typing rule for those primitives.
+*)
+let translucid_apply_prim ~rev typ =
```

Comment:
This name is a bit opaque. Maybe just `check_apply_prim_type`. Also having a type:
```ocaml
type apply_prim =
  | Apply
  | Revapply
```
and using it in place of the `rev` parameter would probably make the code slightly clearer.

Reply:
Being extra-explicit on this very ad-hoc function function seems like a good idea.